internal/cluster/host_spec_subresource.go

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package cluster
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/network"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+// HostSpecSchema this helper function extracts the Host
+// schema, so that it's made available for both Domain and Cluster creation.
+func HostSpecSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of a vSphere host in the free pool",
```

Comment:
This one and the ones below should be for an "ESXi host".

```suggestion
				Description: "ID of an ESXi host in the free pool",
```